src/build-tools/Cargo.toml

```diff
@@ -6,18 +6,42 @@ edition.workspace = true
 rust-version.workspace = true
 publish = false
 
+[lints]
+workspace = true
+
 [dependencies]
 cfg-if = "1.0.0"
+compile-time-run = "0.2.12"
 protobuf-src = { version = "2.0.0", optional = true }
 which = "4"
 workspace-hack = { version = "0.0.0", path = "../workspace-hack", optional = true }
 
-[lints]
-workspace = true
+[dev-dependencies]
+mz-ore = { path = "../ore", features = ["test"] }
 
 [features]
 default = ["protobuf-src", "workspace-hack"]
 bazel = []
 
 [package.metadata.cargo-udeps.ignore]
 normal = ["workspace-hack"]
+
+[package.metadata.cargo-gazelle.lib]
+features_override = ["default", "bazel"]
+extra_deps = ["@rules_rust//tools/runfiles"]
+compile_data = ["@//:git_version"]
+data = [
+    "@protobuf//:protoc",
+    "@protobuf//:well_known_type_protos",
+]
+rustc_flags = ["--cfg=bazel"]
+
+# Skip generating doc tests because there isn't a way to set the rustc flags
+# used for the test, so we can't set the `--cfg=bazel` flag.
+skip_doc_test = true
+
+[package.metadata.cargo-gazelle.lib.rustc_env]
+MZ_GIT_VERSION_PATH = "$(execpath @//:git_version)"
+
+[package.metadata.cargo-gazelle.test.lib]
+rustc_flags = ["--cfg=bazel"]
```

Comment:
note: these changes are not part of this PR but are a small preview as to how configuring the cargo-gazelle for a crate (https://github.com/MaterializeInc/materialize/pull/27346) will work. No other crate should ever need to be this complex, `build-tools` only is because it helps bridge the gap between Bazel and Cargo.

This config was complex enough that I thought it was worthwhile to check-in for now. If folks prefer I can pull this out and save it somewhere!